### PR TITLE
added archlinux packaging script

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: archlinux/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Maintainer: Augusto Hack <hack.augusto@gmail.com>
+pkgname=solc-static
+pkgver=0.4.1
+pkgrel=1
+pkgdesc='solc compiler statically linked against musl'
+arch=('x86_64')
+url='https://github.com/brainbot-com/solidity-static'
+license=('GPL3')
+conflicts=('ethereum' 'solc' 'ethereum-git')
+source=('https://github.com/brainbot-com/solidity-static/releases/download/v0.4.1/solc')
```
Comment (konradkonrad) [inline archlinux/PKGBUILD:10]: @hackaugusto How do you want to handle versioning? Should we just raise the version strings here with each release?

**Comment by hackaugusto** [comment]
For that `PKGBUILD`, adding manually the `pkgver` is the proper way (since a new version requires changes to the source url.

File: archlinux/PKGBUILD (hunk 1 above)
Comment (hackaugusto) [reply archlinux/PKGBUILD:10]: For that PKGBUILD, adding the pkgver manually is the proper way (since a new version requires changes to the source url).

Comment (konradkonrad) [reply archlinux/PKGBUILD:10]: okay. I think `bumpversion` might be a nice tool here to change the ~3 places for each version raise

Comment (hackaugusto) [reply archlinux/PKGBUILD:10]: If we know that for each tag there will have a corresponding release and that the release url will have the tag name, this is an option:

```
pkgver="0.4.1"
source=("https://github.com/brainbot-com/solidity-static/releases/download/v${pkgver}/solc")
pkgver() {
    git tag --sort version:refname | tail -n1 | sed "s,^v,,"
}

```

Comment (konradkonrad) [reply archlinux/PKGBUILD:10]: huh! I missed that latest response. sounds good, please do this!

Comment (konradkonrad) [reply archlinux/PKGBUILD:10]: Yes, we know that 
- [x] for each tag there will have a corresponding release and that 
- [x] the release url will have the tag name

**Comment by utzig** [comment]
LGTM. Btw, 0.4.2 is already available!

File: archlinux/PKGBUILD (hunk 1 above)
Comment (hackaugusto) [reply archlinux/PKGBUILD:10]: Can we generate a corresponding signature for each solc binary?

> Files in the source array with extensions .sig, .sign or, .asc are recognized by makepkg as PGP signatures and will be automatically used to verify the integrity of the corresponding source file.

**Comment by konradkonrad** [comment]
I will merge and 
- raise the current version
- add a bumpversion configuration for easier future upgrades
  afterwards.

**Pull request merged**